docs/running-locally.md

```diff
@@ -1,23 +1,51 @@
-- You will need to install the node depdendencies. You only need this step once
-
+# Run cardanow locally
+## Setup tooling
+In order to work with the project scripts some tools are (some examples are `cardano-cli` and `mithril-client`). If you run nix, everything will be available in the nix flakes shell, these commands will enable a shell with all the tools that will be in this guide:
 ```
-npm i
+git submodule update --init --recursive
+nix develop
 ```
 
-- Use the `dowload-mithril-snapshot.sh` script to download the snapshot. You will need to specify a NETWORK env var
+If you are not a nix user, you will have to manually install the various tools:
+- docker
+- docker-compose
+- cardano-cli
+- mithril-client-cli
+- jq
+- nodejs
 
-```
-NETWORK=preview ./components/cardano-node/dowload-mithril-snapshot.sh
-```
+**NOTE**: the versions of these tools are not yet bounded, the use case is still small enough that no version constrains are required yet, in the future che versions of at least the `mithril-client` and the `cardano-cli` will be specified.
 
-- Start the compose environment. You will need to specify a NETWORK env var
+## Install nodejs dependencies
+You will need to install the node dependencies. You only need this step once
 
+```shell
+npm i
 ```
-NETWORK=preview docker-compose up -d
+
+## Download Mithril snapshot
+Use the `download-mithril-snapshot.sh` script to download the snapshot. You will need to specify a `NETWORK` env var
+
+```shell
+NETWORK=preview ./components/cardano-node/download-mithril-snapshot.sh
 ```
 
-- Run the application
+## Build environment - isolated network
+Start the compose environment. You will need to specify a `NETWORK` env var:
+```shell
+NETWORK=preview
+MITHRIL_CONFIG=configurations/mithril-configs/$NETWORK.env
+docker compose --env-file $MITHRIL_CONFIG up -d
```

Comment:
Oh thats nice, does this automatically make all the env vars in MITHRIL_CONFIG available to containers?

Reply:
No it does not, with the current setup it seem that's not necessary: the variables defined into the `$MITHRIL_CONFIG` are used inside the interpolations in the docker-compose file... am I missing something, do we need to have those set inside the containers?